mediapipe/graphs/instantmotiontracking/calculators/gl_animation_overlay_calculator.cc

```diff
@@ -591,28 +578,45 @@ ::mediapipe::Status GlAnimationOverlayCalculator::Process(
     }
     GLCHECK(glClear(GL_DEPTH_BUFFER_BIT));
 
-    if (has_occlusion_mask_) {
-      glColorMask(GL_FALSE, GL_FALSE, GL_FALSE, GL_FALSE);
-      const TriangleMesh &mask_frame = mask_meshes_.front();
-      MP_RETURN_IF_ERROR(GlBind(mask_frame, mask_texture_));
-      // Draw objects using our latest model matrix stream packet.
-      for (const ModelMatrix &model_matrix : current_mask_model_matrices_) {
-        MP_RETURN_IF_ERROR(GlRender(mask_frame, model_matrix.get()));
-      }
-    }
-
     glColorMask(GL_TRUE, GL_TRUE, GL_TRUE, GL_TRUE);
+
     int frame_index = GetAnimationFrameIndex(cc->InputTimestamp());
-    const TriangleMesh &current_frame = triangle_meshes_[frame_index];
-    MP_RETURN_IF_ERROR(GlBind(current_frame, texture_));
+
+    // Define current frames for all animation assets
+    TriangleMesh &robot_current_frame = robot_triangle_meshes_[frame_index];
+    TriangleMesh &dino_current_frame = dino_triangle_meshes_[frame_index];
+    TriangleMesh &gif_current_frame = gif_triangle_meshes_[0];
+
+    // Load in the latest GIF frame for dynamic texturing
+    if (cc->Inputs().HasTag("GIF_TEXTURE")) {
+      const auto &input_gif_texture =
+          cc->Inputs().Tag("GIF_TEXTURE").Get<AssetTextureFormat>();
+      gif_texture_ = helper_.CreateSourceTexture(input_gif_texture);
+    }
+
+    // Load the render data to map model matrix to texture and asset
+    const std::vector<int> render_ids = cc->Inputs().Tag("RENDER_DATA").Get<std::vector<int>>();
+
     if (has_model_matrix_stream_) {
-      // Draw objects using our latest model matrix stream packet.
+      int idx = 0;
       for (const ModelMatrix &model_matrix : current_model_matrices_) {
```

Comment:
It's a lot more efficient for the rendering if we can avoid switching back and forth between object types-- so we should refactor this to render all (render_id 0)'s first, then all (render_id 1)'s, etc.  Then we end up calling GlBind only once per render_id.